pytest_inmanta/plugin.py

```diff
@@ -293,19 +301,52 @@ def project_metadata(request: pytest.FixtureRequest) -> module.ProjectMetadata:
         )
     )
 
-    index_urls: Sequence[str] = pip_index_url.resolve(request.config)
+    index_urls: Sequence[str] = parameters.pip_index_url.resolve(request.config)
+    repos_urls: List[str] = [
+        repo["url"]
+        for repo in repos
+        if repo["type"] == module.ModuleRepoType.package.value
+    ]
+
+    pip_use_system_config = parameters.pip_use_system_config.resolve(request.config)
+    pip_pre = parameters.pip_pre.resolve(request.config)
+
     modulepath = ["libs"]
     in_place = inm_mod_in_place.resolve(request.config)
     if in_place:
         modulepath.append(str(Path(CURDIR).parent))
 
-    if SUPPORTS_PROJECT_PIP_INDEX:
+    if SUPPORTS_PROJECT_PIP_INDEX_ISO7:
+        # Backward compat: translate repo url to index url
+        index_urls = list(index_urls) + repos_urls
+        if index_urls:
+            index_url = index_urls[0]
+            extra_index_url = index_urls[1:]
+        else:
+            index_url = None
+            extra_index_url = []
+
+        pip_config: ProjectPipConfig = ProjectPipConfig(
+            index_url=index_url,
+            extra_index_url=extra_index_url,
+            use_system_config=pip_use_system_config,
+            pre=pip_pre,
+        )
+
+        if not pip_config.has_source():
+            LOGGER.warning(PIP_NO_SOURCE_WARNING)
+
+        return module.ProjectMetadata(
+            name="testcase",
+            description="Project for testcase",
+            repo=repos,
```

Comment:
Passing on the package type repos to core will trigger a warning. We used to do this because core would still use it for backwards compatibility, but that is no longer the case. So I think we should only pass on the v1 repos, and update our warning message in this case (it states that the compiler will raise a warning).

Reply:
I don't see why we need to update this?

Here, it still works in a backward compatible way and the compile will still raise a warning ?

Reply:
The compiler ignores it now, doesn't it?

Reply:
I mean, it still raises a warning, but it is otherwise ignored.

Reply:
Yes, but we convert them into pip config here as well.

Reply:
Yes, but we still trigger the warning from core, which is not tailored to pytest-inmanta usage.

Reply:
done